cmd/juju/controller/configcommand.go

```diff
@@ -224,7 +224,37 @@ func (c *configCommand) setConfig(client controllerAPI, ctx *cmd.Context) error
 	if err != nil {
 		return errors.Trace(err)
 	}
-	return errors.Trace(client.ConfigSet(attrs))
+
+	store := c.ClientStore()
+	controllerName, err := store.CurrentController()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	ctrl, err := store.ControllerByName(controllerName)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	coerced, err := controller.NewConfig(ctrl.ControllerUUID, ctrl.CACert, attrs)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	extraValues := set.NewStrings()
+	values := make(map[string]interface{})
+	for k := range attrs {
+		extraValues.Add(k)
```

Comment:
if we move `extraValues.Add(k)` to else block then no need the `extraValues.Remove(k)` anymore

```go
if controller.AllowedUpdateConfigAttributes.Contains(k) {
    values[k] = coerced[k]
} else {
    extraValues.Add(k)
}
```

Reply:
good catch, ty